.github/workflows/ci_tests.yml

```diff
@@ -31,33 +31,32 @@ jobs:
             platform: "windows-latest"
           - preset: "msvc-release"
             platform: "windows-latest"
-    name: "Preset: ${{ matrix.presets.preset }} on ${{ matrix.presets.platform }}"
-    runs-on: ${{ matrix.presets.platform }}
+    name: "Preset: ${{ matrix.presets.preset }} on ${{ matrix.presets.platform || matrix.presets.compiler }}"
+    runs-on: ${{ matrix.presets.platform || 'ubuntu-latest' }}
+    container:
+      image: ${{ matrix.presets.compiler && 'ghcr.io/bemanproject/testingcontainers-' }}${{ matrix.presets.compiler }}
     steps:
       - uses: actions/checkout@v4
-      - name: Setup build environment
-        uses: lukka/get-cmake@latest
+      - name: Setup Environment
+        if: ${{ !matrix.presets.compiler }}
+        uses: ./.github/actions/setup-enviornment
         with:
           cmakeVersion: "~3.25.0"
```

Comment:
As noted in the scope reviews, 3.25 is insufficient to support modules 3.28 is required: https://github.com/bemanproject/scope/pull/34/files

Reply:
Concur, this is just to get the compilers on the right versions